crates/hyperqueue/src/dashboard/ui/screens/worker/worker_tasks_table.rs

```diff
@@ -0,0 +1,100 @@
+use crate::dashboard::data::tasks_timeline::TaskInfo;
+use crate::dashboard::ui::terminal::DashboardFrame;
+use crate::dashboard::ui::widgets::table::{StatefulTable, TableColumnHeaders};
+use humantime::format_duration;
+use std::time::SystemTime;
+use tako::TaskId;
+use tui::layout::{Constraint, Rect};
+use tui::widgets::{Cell, Row};
+
+#[derive(Default)]
+pub struct WorkerTasksTable {
+    table: StatefulTable<WorkerTaskRow>,
+}
+
+impl WorkerTasksTable {
+    pub fn update(&mut self, tasks_info: Vec<&TaskInfo>) {
+        let rows = create_rows(tasks_info);
+        self.table.set_items(rows);
+    }
+
+    pub fn select_next_task(&mut self) {
+        self.table.select_next_wrap();
+    }
+
+    pub fn select_previous_task(&mut self) {
+        self.table.select_previous_wrap();
+    }
+
+    pub fn get_selected_item(&self) -> Option<TaskId> {
+        let selection = self.table.current_selection();
+        selection.map(|row| row.task_id)
+    }
+
+    pub fn draw(&mut self, rect: Rect, frame: &mut DashboardFrame) {
+        self.table.draw(
+            rect,
+            frame,
+            TableColumnHeaders {
+                title: "Tasks Info".to_string(),
+                inline_help: "".to_string(),
+                table_headers: Some(vec!["Task ID", "State", "Start", "Makespan", "End"]),
+                column_widths: vec![
+                    Constraint::Percentage(20),
+                    Constraint::Percentage(20),
+                    Constraint::Percentage(20),
+                    Constraint::Percentage(20),
+                    Constraint::Percentage(20),
+                ],
+            },
+            |task_row| {
+                Row::new(vec![
+                    Cell::from(task_row.task_id.to_string()),
+                    Cell::from(task_row.task_state.clone()),
+                    Cell::from(task_row.start_time.clone()),
+                    Cell::from(task_row.run_time.clone()),
```

Comment:
Same with duration, I suppose to only show minutes and seconds, and if the duration is smaller than a minute, then also include milliseconds.